addons/sale/models/product_product.py

```diff
@@ -1,23 +1,26 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, fields, models
+from odoo import fields, models
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
-    @api.multi
+    sales_count = fields.Integer(compute='_sales_count', string='# Sales')
+
     def _sales_count(self):
-        r = {}
-        domain = [
-            ('state', 'in', ['sale', 'done']),
-            ('product_id', 'in', self.ids),
-        ]
-        for group in self.env['sale.report'].read_group(domain, ['product_id', 'product_uom_qty'], ['product_id']):
-            r[group['product_id'][0]] = group['product_uom_qty']
+        self._cr.execute("""
+        SELECT  l.product_id,
+        sum(l.product_uom_qty / u.factor * u2.factor) AS product_uom_qty                    
+        FROM sale_order_line l
+           JOIN sale_order s ON l.order_id = s.id     
+           LEFT JOIN product_product p ON l.product_id = p.id
+           LEFT JOIN product_template t ON p.product_tmpl_id = t.id
+           LEFT JOIN product_uom u ON u.id = l.product_uom
+           LEFT JOIN product_uom u2 ON u2.id = t.uom_id
+        WHERE s.state in ('done', 'sale') and product_id in ({product_ids})
+        GROUP BY l.product_id""".format(product_ids=", ".join(str(i) for i in self.ids)))
```

Comment:
```
query = """
 SELECT  l.product_id,
sum(l.product_uom_qty / u.factor * u2.factor) AS product_uom_qty                          
FROM sale_order_line l
JOIN sale_order s ON l.order_id = s.id     
LEFT JOIN product_product p ON l.product_id = p.id
LEFT JOIN product_template t ON p.product_tmpl_id = t.id
           LEFT JOIN product_uom u ON u.id = l.product_uom
           LEFT JOIN product_uom u2 ON u2.id = t.uom_id
        WHERE s.state in ('done', 'sale') and product_id in %s
        GROUP BY l.product_id"""
params = (tuple(self.ids),)
self._cr.execute(query, params)
```

Use params argument to prevent SQL injection

Reply:
Changed it, thx for comment!